open System
(*
  Ce test a été généré par Metalang.
*)
exception Found_1 of int

let result len (tab:int array) =
  try
  let tab2 = Array.init len (fun _i ->
    false)
  for i1 = 0 to len - 1 do
    Printf.printf "%d " tab.[i1]
    tab2.[tab.[i1]] <- true
  Printf.printf "\n"
  for i2 = 0 to len - 1 do
    if (not tab2.[i2]) then
      raise (Found_1(i2))
  (- 1)
  with Found_1 (out) -> out

let () =
  let len = (Console.ReadLine () |> Int32.Parse)
  Printf.printf "%d\n" len
  let tab = ((Console.ReadLine ()).Split [|' '|] |> Array.map Int32.Parse)
  Printf.printf "%d\n" (result len tab)
 